Fix doc (#1767)
huggingface/datasets-server: docs/source/search.mdx

# Search text in a dataset

Datasets Server provides a `/search` endpoint for searching words in a dataset.

<Tip warning={true}>
  Currently, only {" "}

>   <a href="./parquet">datasets with parquet exports</a>

Reviewer: ```suggestion
  <a href="./parquet">datasets with Parquet exports</a>
```

  are supported so Datasets Server can index the contents and run the search without downloading the
  whole dataset.
</Tip>

This guide shows you how to use Datasets Server's `/search` endpoint to search for a query string.
Feel free to also try it out with [ReDoc](https://redocly.github.io/redoc/?url=https://datasets-server.huggingface.co/openapi.json#operation/searchRows).

The text is searched in the columns of type `string` or `ClassLabel`, even if the values are nested in a dictionary.

The `/search` endpoint accepts five query parameters:

- `dataset`: the dataset name, for example `glue` or `mozilla-foundation/common_voice_10_0`
- `config`: the configuration name, for example `cola`
- `split`: the split name, for example `train`
- `query`: the text to search
- `offset`: the offset of the slice, for example `150`
- `length`: the length of the slice, for example `10` (maximum: `100`)

> 

Reviewer: I think here it would be nice to explicitly say what we're searching for here (can remove the mention of this later in the doc). Something like:

"For example, let's search for the text `dog` in the `train` split of the `SelfRC` configuration of the `duorc` dataset from slice 150-151:"

def query():
    response = requests.get(API_URL, headers=headers)
    return response.json()
data = query()
```
</python>
<js>
```js
import fetch from "node-fetch";
async function query(data) {
    const response = await fetch(
        "https://datasets-server.huggingface.co/search?dataset=duorc&config=SelfRC&split=train&query=dog&offset=150&length=2",
        {
            headers: { Authorization: `Bearer ${API_TOKEN}` },
            method: "GET"
        }
    );
    const result = await response.json();
    return result;
}
query().then((response) => {
    console.log(JSON.stringify(response));
});
```
</js>
<curl>
```curl
curl https://datasets-server.huggingface.co/search?dataset=duorc&config=SelfRC&split=train&query=dog&offset=150&length=2 \
        -X GET \
        -H "Authorization: Bearer ${API_TOKEN}"
```
</curl>
</inferencesnippet>

The endpoint response is a JSON containing two keys (same format as [`/rows`](./rows)):

- The [`features`](https://huggingface.co/docs/datasets/about_dataset_features) of a dataset, including the column's name and data type.
- The slice of `rows` of a dataset and the content contained in each column of a specific row.

> The rows are ordered by the row index. Note that there is no indication of the pertinence of a specific row, and the matching strings are not highlighted.

Reviewer: ```suggestion
The rows are ordered by the row index, and the text strings matching the query are not highlighted.
```
